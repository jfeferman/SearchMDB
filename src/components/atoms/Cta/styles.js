import styled from 'styled-components';
import { px2rem, colors } from '../../styleUtils/globalStyleVariables';

export const StyledButton = styled.button`
  border-radius: ${px2rem(10)};
  border: 1px solid;
  cursor: pointer;
  display: block;
  font-weight: 300;
  text-align: center;
  text-decoration: none;
  white-space: nowrap;
  width: auto;
  background: ${colors.lightGray};
  color: ${colors.black};
  font-size: ${px2rem(15)};
  line-height: ${px2rem(17)};
  letter-spacing: ${px2rem(0.34)};
  padding: ${px2rem(16)} ${px2rem(41.5)};
  vertical-align: middle;
  outline: 0;
  transition: color .15s ease-in-out,
    background-color .15s ease-in-out,
    border-color .15s ease-in-out;

  &:focus {
    box-shadow: 0px 0px 2px 2px rgba(23,109,220,0.8);
  }

  ${(props) => props.disabled && `
    cursor: not-allowed;
    background: ${colors.disabled};
    color: ${colors.disabledText};
    border: 0;
    &:hover {
      color: ${colors.disabledText};
    }
  `};
`;
