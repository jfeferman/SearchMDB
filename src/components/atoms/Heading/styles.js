import styled from 'styled-components';
import { fonts } from '../../styleUtils/globalStyleVariables';

export const StyledH1 = styled.h1`
  ${fonts.H1}
`;

export const StyledH2 = styled.h2`
  ${fonts.H2}
`;

export const StyledH3 = styled.h3`
  ${fonts.H3}
`;

export const StyledH4 = styled.h4`
  ${fonts.H4}
`;
