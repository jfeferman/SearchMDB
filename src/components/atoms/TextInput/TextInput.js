import React from 'react';
import PropTypes from 'prop-types';
import {
  StyledTextInput,
} from './styles';

const TextInput = (props) => {
  const {
    placeholder,
    onChange,
  } = props;

  return (
    <StyledTextInput
      type="text"
      placeholder={placeholder}
      onChange={onChange}
    />
  );
};

TextInput.propTypes = {
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
};

export default TextInput;
