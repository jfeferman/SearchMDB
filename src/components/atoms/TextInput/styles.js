import styled from 'styled-components';
import { px2rem, colors } from '../../styleUtils/globalStyleVariables';

export const StyledTextInput = styled.input`
  width: 100%;
  box-sizing: border-box;
  height: ${px2rem(56)};
  border: 1px solid ${colors.lightGray};
  border-radius: ${px2rem(3)};
  background-color: ${colors.white};
  padding-left: ${px2rem(20)};
  font-size: ${px2rem(16)};
`;
