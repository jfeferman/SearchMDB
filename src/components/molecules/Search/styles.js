import styled from 'styled-components';
import { px2rem } from '../../styleUtils/globalStyleVariables';

export const StyledSearch = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: ${px2rem(600)};

  div:nth-of-type(1) {
    margin-right: ${px2rem(10)};
  }
`;
