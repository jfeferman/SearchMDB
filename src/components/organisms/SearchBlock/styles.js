import styled from 'styled-components';
import { px2rem } from '../../styleUtils/globalStyleVariables';

export const StyledSearchBlock = styled.div`
  margin: ${px2rem(100)} auto ${px2rem(100)} auto;
  
  h1 {
    text-align: center;
  }
`;
