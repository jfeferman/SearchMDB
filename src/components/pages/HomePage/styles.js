import styled from 'styled-components';
import { px2rem } from '../../styleUtils/globalStyleVariables';

export const StyledResultsBlock = styled.div`
  margin: 0 auto;
  max-width: ${px2rem(980)};
  padding: 0 ${px2rem(40)} ${px2rem(80)} ${px2rem(40)};

  h3 {
    margin-bottom: ${px2rem(4)};
  }
`;
