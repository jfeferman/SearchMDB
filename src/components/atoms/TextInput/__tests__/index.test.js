import React from 'react';
import { create, act } from 'react-test-renderer';
import TextInput from '..';
import { StyledTextInput } from '../styles';

describe('components/atoms/TextInput', () => {
  let renderer;
  let instance;
  let input;
  const placeholder = 'Placeholder text';
  const onChange = jest.fn();

  beforeEach(() => {
    renderer = create(
      <TextInput
        placeholder={placeholder}
        onChange={onChange}
      />,
    );
    instance = renderer.root;
    input = instance.find(el => el.type === StyledTextInput);
  });

  it('matches the snapshot of the component', () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('verifies that a styled text input component is present', () => {
    expect(instance.findAll(el => el.type === StyledTextInput)).toHaveLength(1);
  });

  it('verifies the placeholder text', () => {
    expect(instance.props.placeholder).toBe(placeholder);
  });

  it('verifies function is called when input is changed', () => {
    act(() => {
      input.props.onChange();
    });
    expect(onChange).toHaveBeenCalled();
  });
});
