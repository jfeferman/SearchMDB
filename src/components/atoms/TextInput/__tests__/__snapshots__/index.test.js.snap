// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library/atoms/TestSaveInput matches the snapshot of the component 1`] = `
<input
  className="sc-bdVaJa hSPkxn"
  onChange={[MockFunction]}
  placeholder="Placeholder text"
  type="text"
/>
`;
